From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Fri, 12 Nov 2021 23:58 +0000
Subject: [PATCH] mkinitfs: Handle compressed kernel modules

When determining which firmware needs to be includes in the initramfs
mkinitfs searchs for kernel modules using the filename *.ko and so
does not cater for compressed modules (and therefore does not include
their required firmware in the initramfs).

Change the search pattern to deal with both uncompressed and
compressed modules.

---

diff -aur a/mkinitfs.in b/mkinitfs.in
--- a/mkinitfs.in
+++ b/mkinitfs.in
@@ -138,9 +138,9 @@
 	rm -rf "$tmpdir"/lib/firmware
 	mkdir -p "$tmpdir"/lib/firmware
 	# Verify if there are initfs modules
-	_modules=`find "$tmpdir"/lib/modules -type f -name "*.ko" -print -quit`
+	_modules=`find "$tmpdir"/lib/modules -type f -name "*.ko*" -print -quit`
 	[ -n "$_modules" ] || return 0
-	find "$tmpdir"/lib/modules -type f -name "*.ko" | xargs modinfo -F firmware | sort -u | while read FW; do
+	find "$tmpdir"/lib/modules -type f -name "*.ko*" | xargs modinfo -F firmware | sort -u | while read FW; do
 		[ -e "${basedir}/lib/firmware/${FW}" ] && install -pD "${basedir}/lib/firmware/${FW}" "$tmpdir"/lib/firmware/$FW
 	done
 	return 0
